visualization/src/main/kotlin/org/jetbrains/kotlinx/dl/visualization/letsplot/PlotConv2D.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2021 JetBrains s.r.o. and Kotlin Deep Learning project contributors. All Rights Reserved.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlinx.dl.visualization.letsplot
+
+import jetbrains.letsPlot.GGBunch
+import org.jetbrains.kotlinx.dl.api.core.TrainableModel
+import org.jetbrains.kotlinx.dl.api.core.layer.convolutional.Conv2D
+
+internal val FILTER_LAYERS_PERMUTATION = intArrayOf(1, 0, 2, 3)
+
+internal val ACTIVATION_LAYERS_PERMUTATION = intArrayOf(2, 1, 0, 3)
+
+fun filtersPlot(
+    conv2DLayer: Conv2D,
+    plotFill: PlotFill = PlotFill.GRAY,
+    imageSize: Int = 64,
+    columns: Int = 8
+): GGBunch {
+    @Suppress("UNCHECKED_CAST")
+    val weights = conv2DLayer.weights.values.toTypedArray()[0] as TensorImageData
+
+    val xyio = extractXYIOSizes(weights, FILTER_LAYERS_PERMUTATION)
+
+    val plots = inputsOutputsPlots(xyio[2], xyio[3]) { i, o ->
+        xyPlot(xyio[0], xyio[1], plotFill) { x, y ->
+            weights[y][x][i][o]
+        }
+    }
+    return columnPlot(plots, columns, imageSize)
+}
+
+fun modelActivationOnLayersPlot(
+    model: TrainableModel,
+    x: FloatArray,
+    plotFill: PlotFill = PlotFill.GRAY,
+    imageSize: Int = 64,
+    columns: Int = 8,
+): List<GGBunch> {
+    val activations = model.predictAndGetActivations(x).second
+    @Suppress("UNCHECKED_CAST")
+    val activationArrays = activations.mapNotNull { it as? TensorImageData }
+
+    return activationArrays.map { weights ->
+        val xyio = extractXYIOSizes(weights, ACTIVATION_LAYERS_PERMUTATION)
```

Comment:
could you please rename this variable (xyio) it reminds me not a very decent word

Reply:
it's for:
x axis
y axis
i put
o utput
Can chnage it to full words if you wish (I thought that this approach is more readable but now I see that only for me 😄 )

Reply:
will change it of course 😉